Makefile

```diff
@@ -1,11 +1,22 @@
-version:
+.PHONY: version
+version: ## Bump version and push to release branch.
+	ifndef v
+		$(error v is required, ex. make version v=0.1.0)
 	@poetry version $(v)
 	@git add pyproject.toml
 	@git commit -m "Version bump v$$(poetry version -s)"
 	@git push origin release/v$$(poetry version -s)
 
-publish:
+.PHONY: publish
+publish: ## Publish version to Git and PyPI.
 	@git tag v$$(poetry version -s)
 	@git push --tags
 	@poetry build
 	@poetry publish
+
+.DEFAULT_GOAL := help
+.PHONY: help
+help: ## Print Makefile help text.
+	@grep -E '^[a-zA-Z_\/-]+:.*?## .*$$' $(MAKEFILE_LIST) \
+	| awk 'BEGIN {FS = ":.*?## "}; \
+		{printf "\033[36m%-20s\033[0m%s\n", $$1, $$2}'
```

Comment:
Can you explain this wizardry a bit 😅

Reply:
This uses grep to parse the Makefile for targets with comments in the format:

```
target: ## comment
```

These are passed to awk which then splits the incoming line into a target and comment variable, then prints them inside some terminal formatting directives. Any targets without comments are ignored and excluded from help text.